app/controllers/facility_journals_controller.rb

```diff
@@ -30,12 +30,17 @@ def new_with_search
     set_default_variables
     @layout = "two_column_head"
 
-    respond_to do |format|
-      format.csv do
-        # used for "Export as CSV" link for order details with expired accounts
-        @order_details = @order_details.expired_account
+    @after_search_hook = lambda do
+      @valid_order_details, @invalid_order_details = ValidatorFactory.partition_valid_order_details(@order_details.unexpired_account)
+      @invalid_order_details += @order_details.expired_account
```

Comment:
We can sort them, but it looks like the basic scope (`.need_journal`) doesn't do any ordering so I'm not sure what would make sense to order by.  There can't be duplicates, since they're disjoint sets.